src/backend/distributed/executor/executor.c

```diff
@@ -1183,25 +1132,45 @@ ManageWorkerPool(WorkerPool *workerPool)
 		return;
 	}
 
-	if (currentConnectionCount >= targetPoolSize)
+	if (UseSeparateConnectionPerPlacement())
 	{
-		/* already reached the minimal pool size */
-		return;
+		int unusedConnectionCount = workerPool->unusedConnectionCount;
```

Comment:
Makes sense, though as a counter point: it introduces a new constraint that `ManageWorkerPool` always finishes at once. In the future, we'll introduce things like 10ms space between connection attempts. We probably would not want to do that in the force case, but it'll make the code slightly harder to reason about if we make different assumptions in different code paths.

Reply:
Discussed this off-line, but this would complicate the logic for failover as we may suddenly have a greater number of ready tasks. It would also require a separate flag, because `initiatedConnectionCount` may be greater than 0 from the beginning if there is a separate task.